pkg/build/api/v1beta3/types.go

```diff
@@ -382,6 +386,21 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty" description:"forces the source build to pull the image if true"`
 }
 
+// BuildPostCommitSpec holds a build post commit hook specification.
+type BuildPostCommitSpec struct {
+	// Shell is an action that executes a shell script in a container
+	// running the build output image.
+	Shell string `json:"shell,omitempty" description:"shell script to be executed in a container running the build output image"`
```

Comment:
> ShellArgs must be an array - otherwise there is no way for callers to guarantee that that proper quoting is preserved across all runtimes

Hmm, I think I convinced @bparees just a while ago that there's no benefit wrt quoting if `ShellArgs` is an array of strings:

```console
$ make test | grep -v 'FAIL("all of Bob'\''s tests")'
```

```json
{
  "ShellArgs": ["make test | grep -v 'FAIL(\"all of Bob'\\''s tests\")'"]
}
```

How is `["arg1", "arg2", "arg3"]` helping with quoting if in the end of the day all we do is concatenate with spaces?
`["arg1 arg2 arg3"]`

The JSON below, although valid, won't produce the right outcome:

```json
{
  "ShellArgs": [
    "make test | grep -v 'FAIL(",
    "all of Bob'\\''s tests",
    ")'"
  ]
}
```